dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -53,58 +67,168 @@ class DataManager(
     }
 
     /**
-     * Method to make the data manager add data to a data store and store meta data in Dataland
+     * Method to make the data manager add data to a data store, store metadata in Dataland and sending messages to the
+     * relevant message queues
      * @param storableDataSet contains all the inputs needed by Dataland
      * @return ID of the newly stored data in the data store
      */
     @Transactional
-    fun addDataSet(storableDataSet: StorableDataSet, correlationId: String): String {
+    fun addDataSetToTemporaryStorageAndSendMessage(storableDataSet: StorableDataSet, correlationId: String):
+        String {
         val company = companyManager.getCompanyById(storableDataSet.companyId)
         logger.info(
             "Sending StorableDataSet of type ${storableDataSet.dataType} for company ID " +
                 "${storableDataSet.companyId}, Company Name ${company.companyName} to storage Interface. " +
                 "Correlation ID: $correlationId",
         )
-        val dataId: String = storeDataSet(storableDataSet, company.companyName, correlationId)
-        metaDataManager.storeDataMetaInformation(
-            dataId,
-            storableDataSet.dataType,
-            storableDataSet.uploaderUserId,
-            storableDataSet.uploadTime,
-            company,
+        val dataId: String = storeDataSetInTemporaryStoreAndSendMessage(
+            storableDataSet, company.companyName, correlationId,
+        )
+        val metaData = DataMetaInformationEntity(
+            dataId, storableDataSet.dataType.toString(),
+            storableDataSet.uploaderUserId, storableDataSet.uploadTime, company, QAStatus.Pending,
         )
+        metaDataManager.storeDataMetaInformation(metaData)
         return dataId
     }
 
-    private fun storeDataSet(
+    /**
+     * Method that listens to the qa_queue and updates the metadata information after successful qa process
+     * @param jsonString the message describing the result of the completed QA process
+     * @param correlationId the correlation ID of the current user process
+     * @param type the type of the message
+     */
+    @RabbitListener(
+        bindings = [
+            QueueBinding(
+                value = Queue("dataQualityAssuredBackendDataManager"),
+                exchange = Exchange(ExchangeNames.dataQualityAssured, declare = "false"),
+                key = [""],
+            ),
+        ],
+    )
+    fun updateMetaData(
+        @Payload jsonString: String,
+        @Header(MessageHeaderKey.CorrelationId) correlationId: String,
+        @Header(MessageHeaderKey.Type) type: String,
+    ) {
+            if (type != MessageType.QACompleted) {
+//                TODO: Reject statement
+                return
+            }
+        val dataId = objectMapper.readValue(jsonString,QaCompletedMessage::class.java).dataId
+        if (dataId.isNotEmpty()) {
```

Comment:
Can't you simply do:
Instead of:
```
    fun updateMetaData(
        @Payload jsonString: String,
        @Header(MessageHeaderKey.CorrelationId) correlationId: String,
        @Header(MessageHeaderKey.Type) type: String,
    ) {
            if (type != MessageType.QACompleted) {
//                TODO: Reject statement
                return
            }
        val dataId = objectMapper.readValue(jsonString,QaCompletedMessage::class.java).dataId
```

```
    fun updateMetaData(
        @Payload qaCompletedMessage: QaCompletedMessage,
        @Header(MessageHeaderKey.CorrelationId) correlationId: String,
        @Header(MessageHeaderKey.Type) type: String,
    ) {
            if (type != MessageType.QACompleted) {
//                TODO: Reject statement
                return
            }
        val dataId = qaCompletedMessage.dataId
```

Reply:
Does not work out of the box but runs into an error that the Byte array cannot be converted into a Generic message. I think that we'll need to "teach" RabbitMQ that it is communicating via JSON messages to ensure that it would implicitly deserialize JSONs into object. Certainly possible with the right annotations and by customizing the Rabbit Template as e.g. showcased here: https://thepracticaldeveloper.com/produce-and-consume-json-messages-with-spring-boot-amqp/
We would then also not need to convert to JSON when sending out a message.

Therefore this will be a bit more work and critically, we're currently only communicating via JSON at one point while our other Payloads are just a string of the dataid (no curly braces and identifier). I also fear that this might clash or overlap with the CloudEvent stuff and we'll need to do a bit more research to do things "right" considering our usage of the cloud event message handler.

Overall I don't feel like investing this time right now would be correct considering the "sprint goal" of having a completed MVP soon. Switching all RabbitMQ message Payloads to be JSONs and having implicit serialization of messages can also easily be put into a separate ticket instead.